overlord/ifacestate/apparmorprompting/common/common.go

```diff
@@ -0,0 +1,466 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package common
+
+import (
+	"encoding/base32"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/snapcore/snapd/sandbox/apparmor/notify"
+	"github.com/snapcore/snapd/strutil"
+)
```

Comment:
this package doesn't depend on overlord/state so it's unclear why it would live here, also the name is a bit too long. Also it seems that some helpers might belong to timeutil or snap/naming. Most of the rest should go to a subpackage of interfaces, like interfaces/prompting perhaps?

Reply:
Very good point. This also makes much more sense when we want to expose over the API which abstract permissions are available for a given interface.